Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/RenamingPolicy.cs

```diff
@@ -1,270 +1,249 @@
-            }
+        var sourceFileName = Path.GetFileName(sourceFilePath);
+        return Path.Combine(directoryName, sourceFileName);
+    }
 
+    private static string CanonizeAndCheckPath(string path)
+    {
+        path = path.Replace(Path.DirectorySeparatorChar, '/'); // make all the paths use forward slashes - this should be supported on File, FTP, and SFTP
 
-            directoryName = CanonizeAndCheckPath(directoryName);
+        if (path.IndexOfAny(GetInvalidChars()) != -1)
+            throw new ArgumentException("Illegal characters in path: " + path);
+        return path;
+    }
 
-            // this should always be a directory
-            if (!directoryName.EndsWith("/"))
-            {
-                directoryName = directoryName + "/";
-            }
-            var sourceFileName = Path.GetFileName(sourceFilePath);
-            return Path.Combine(directoryName, sourceFileName);
-        }
+    private static char[] GetInvalidChars()
+    {
+        List<char> invalidCharacters = new List<char>(Path.GetInvalidFileNameChars());
+        invalidCharacters.Remove('/'); // remove the forward slash, as it is supported
+        invalidCharacters.Remove(':'); // also the colon is supported
+        return invalidCharacters.ToArray();
+    }
 
-        private static string CanonizeAndCheckPath(string path)
-        {
-            path = path.Replace(Path.DirectorySeparatorChar, '/'); // make all the paths use forward slashes - this should be supported on File, FTP, and SFTP
+    public string CreateRemoteFileNameForRename(string originalFilePath, string sourceOperationTo)
+    {
+        if (String.IsNullOrEmpty(sourceOperationTo))
+            throw new ArgumentException("When using rename as a source operation, you need to define the new name");
 
-            if (path.IndexOfAny(GetInvalidChars()) != -1)
-                throw new ArgumentException("Illegal characters in path: " + path);
-            return path;
-        }
+        string filePath = sourceOperationTo;
+        filePath = ExpandMacrosAndMasks(originalFilePath, filePath);
 
-        private static char[] GetInvalidChars()
-        {
-            List<char> invalidCharacters = new List<char>(Path.GetInvalidFileNameChars());
-            invalidCharacters.Remove('/'); // remove the forward slash, as it is supported
-            invalidCharacters.Remove(':'); // also the colon is supported
-            return invalidCharacters.ToArray();
-        }
+        return CanonizeAndCheckPath(filePath);
+    }
 
-        public string CreateRemoteFileNameForRename(string originalFilePath, string sourceOperationTo)
-        {
-            if (String.IsNullOrEmpty(sourceOperationTo))
-                throw new ArgumentException("When using rename as a source operation, you need to define the new name");
+    public bool IsMacro(string macro)
```

Comment:
You are not using this method, so it can be removed.

Reply:
Removed.